app/services/qualified_teachers_api_sender.rb

```diff
@@ -0,0 +1,92 @@
+class QualifiedTeachersAPISender
+  include ActiveModel::Model
+  include ActiveModel::Attributes
+
+  SUCCESS_CODES = %w[204].freeze
+
+  attribute :participant_outcome_id
+
+  validates :participant_outcome_id, presence: true
+  validates :participant_outcome, presence: true
+  validate :not_already_sent
+
+  def send_record
+    set_sent_to_qualified_teachers_api_at
+
+    return if invalid?
+
+    create_participant_outcome_api_request!
+    set_qualified_teachers_api_request_successful
+    participant_outcome
+  end
+
+  def participant_outcome
+    @participant_outcome ||= ParticipantOutcome.includes(declaration: { application: :user }).find_by(id: participant_outcome_id)
+  end
+
+private
+
+  def not_already_sent
+    return if participant_outcome&.qualified_teachers_api_request_successful.nil?
+
+    errors.add(:participant_outcome, :already_successfully_sent_to_api) if participant_outcome&.qualified_teachers_api_request_successful?
+    errors.add(:participant_outcome, :already_unsuccessfully_sent_to_api) if participant_outcome&.qualified_teachers_api_request_successful == false
+  end
+
+  def set_sent_to_qualified_teachers_api_at
+    participant_outcome.update(
+      sent_to_qualified_teachers_api_at: Time.zone.now,
+    )
+  end
+
+  def create_participant_outcome_api_request!
+    participant_outcome.participant_outcome_api_requests.create!(
+      request_path: api_response.response.uri.to_s,
+      status_code: api_response.response.code,
+      request_headers: api_response.request.each_header.to_h.except("authorization"),
+      request_body: request_body.stringify_keys,
+      response_headers: api_response.response.each_header.to_h,
+      response_body: response_body(api_response.response.body),
+    )
+  rescue StandardError => e
```

Comment:
Can we rescue something more specific here and cover it with a test?

Reply:
It's hard that one as there's a bunch of different exceptions that can happen with the client that does requests using `net/http`.. so we decided to rescue all exceptions instead..